### PR TITLE
updates to tax logic

### PR DESCRIPTION
**Pull request by kangnam**

### CODE REVIEW

File: model/service/TaxService.cfc [hunk 1]
```diff
@@ -55,10 +55,10 @@ component extends="HibachiService" persistent="false" accessors="true" output="f
 
 	public void function updateOrderAmountsWithTaxes(required any order) {
 
-		if (!arguments.order.hasOrderItem()){
```
Comment (rmarchandTen24) [inline model/service/TaxService.cfc]: @ian-a-hickey @kangnam if the cart is emptied we should clear all the taxes. not sure why this was removed.

File: model/service/TaxService.cfc [hunk 2]
```diff
@@ -151,7 +151,9 @@ component extends="HibachiService" persistent="false" accessors="true" output="f
 			for(var orderItem in arguments.order.getOrderItems()) {
 	
 				// Apply Tax for sale items
-				if(orderItem.getOrderItemType().getSystemCode() == "oitSale") {
+				if(orderItem.getOrderItemType().getSystemCode() == "oitSale" && arguments.order.getPaymentAmountDueAfterGiftCards() > 0) {
+	
+					removeTaxesFromOrderItem(orderItem);
```
Comment (rmarchandTen24) [inline model/service/TaxService.cfc]: @ian-a-hickey @kangnam all taxes would have been already removed at this point on line 111. Why are we running this again? If the function is not removing them above you should fix it there

File: model/service/TaxService.cfc (hunk 1 above)
Comment (ian-a-hickey) [reply model/service/TaxService.cfc]: @evan1024 Can you respond to this please?

Comment (rmarchandTen24) [reply model/service/TaxService.cfc]: @evan1024 let's tag up about this. Can you schedule a meeting?

**Comment by evan1024** [comment]
Give me a moment to review. I have a feeling this was Total Wine's custom
code that was accidentally ported to core.

**Pull request merged**